flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSink.java

```diff
@@ -331,7 +338,7 @@ private RowDataPartitionComputer partitionComputer() {
         return new FileSystemFormatFactory.ReaderContext() {
             @Override
             public TableSchema getSchema() {
```

Comment:
I don't think we should, because the `FileSystemFormatFactory` is deprecated and we need to prune it anyway. I was thinking to do that in a separate issue (there are only 2 implementations for that, avro and csv IIRC)